pkg/controller/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -721,33 +727,109 @@ func TestSyncMurStatusWithUserAccountStatuses(t *testing.T) {
 }
 
 func TestDeleteUserAccountViaMasterUserRecordBeingDeleted(t *testing.T) {
-	// given
-	defer counter.Reset()
-	InitializeCounter(t, 2, UserAccountsForCluster(test.MemberClusterName, 2))
-	logf.SetLogger(zap.New(zap.UseDevMode(true)))
-	s := apiScheme(t)
-	mur := murtest.NewMasterUserRecord(t, "john",
-		murtest.Finalizer("finalizer.toolchain.dev.openshift.com"),
-		murtest.ToBeDeleted())
-	userAcc := uatest.NewUserAccountFromMur(mur)
+	t.Run("success", func(t *testing.T) {
+		// given
+		defer counter.Reset()
+		InitializeCounter(t, 2, UserAccountsForCluster(test.MemberClusterName, 2))
+		logf.SetLogger(zap.New(zap.UseDevMode(true)))
+		s := apiScheme(t)
+		mur := murtest.NewMasterUserRecord(t, "john",
+			murtest.ToBeDeleted())
+		userAcc := uatest.NewUserAccountFromMur(mur)
 
-	memberClient := test.NewFakeClient(t, userAcc)
-	hostClient := test.NewFakeClient(t, mur)
+		memberClient := test.NewFakeClient(t, userAcc)
+		hostClient := test.NewFakeClient(t, mur)
 
-	cntrl := newController(t, hostClient, s, NewGetMemberCluster(true, v1.ConditionTrue),
-		ClusterClient(test.MemberClusterName, memberClient))
+		cntrl := newController(t, hostClient, s, NewGetMemberCluster(true, v1.ConditionTrue),
+			ClusterClient(test.MemberClusterName, memberClient))
 
-	// when
-	_, err := cntrl.Reconcile(newMurRequest(mur))
+		// when
+		result1, err1 := cntrl.Reconcile(newMurRequest(mur))
+		require.NoError(t, err1)
+		assert.True(t, result1.Requeue)
```

Comment:
could you please also check that the finalizer is still there after the first reconcile? You could also move the second reconcile-case into a subtest of this test to make it easier to read :-)

Reply:
Done in d203faf29d69fbe2b0a700a162d64ac3ba223dea